### PR TITLE
Add Resend verification email functionality

### PR DESCRIPTION
**Pull request by minhlongdo**

### CODE REVIEW

File: src/main/java/com/auth0/json/mgmt/jobs/Job.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private String status;
+    @JsonProperty("type")
+    private String type;
+    @JsonProperty("created_at")
+    private String createdAt;
+    @JsonProperty("id")
+    private String id;
+
+    @JsonCreator
+    public Job(@JsonProperty("status") String status, @JsonProperty("type") String type, @JsonProperty("id") String id) {
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/json/mgmt/jobs/Job.java]: I think for the time being, this doesn't need to be public. Can you try making it private or package protected at least? See if Jackson complains.

File: src/main/java/com/auth0/json/mgmt/jobs/Job.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return createdAt;
+    }
+
+    @JsonProperty("id")
+    public String getId() {
+        return id;
+    }
+
+    @JsonProperty("created_at")
+    public void setCreatedAt(String createdAt) {
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/json/mgmt/jobs/Job.java]: I don't see any sample values for this `created_at` field in the api docs. But probably would be better to just use a Date and make Jackson convert between the types. I think the API already makes use of the ISO-8601 format. See the User's class created_at method. If this is the case, then you'll also need to update the JSON samples used in the introduced tests.

File: src/main/java/com/auth0/client/mgmt/JobsEntity.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * Sends an Email Verification. A token with scope update:users is needed.
+     * See https://auth0.com/docs/api/management/v2#!/Jobs/post_verification_email
+     *
+     * @param recipient the email verification recipient.
+     * @return a Request to execute.
+     */
+    public Request<Job> sendVerificationEmail(EmailRecipient recipient) {
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/JobsEntity.java]: 2 parameters seems fine to have them expanded on the signature. Why do you suggest to use this new `EmailRecipient` class? Maybe it's more straight forward to validate the user id value (non-null) and let the client id one be optional (nullable). Java-docs would clarify each one's validation respectively.

**Comment by lbalmaceda** [comment]
@minhlongdo are you still interested on this?

**Comment by minhlongdo** [comment]
@lbalmaceda Yes, I am looking at it now. Sorry for the delay.

**Comment by minhlongdo** [comment]
@lbalmaceda could you try to kick off the CI again please? It compiles on my laptop. Thanks! :)
```
minhlongsmbp2:auth0-java minhlongdo$ ./gradlew test
Using version 1.6.0-SNAPSHOT for auth0
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:compileTestJava UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:test UP-TO-DATE

BUILD SUCCESSFUL

Total time: 4.151 secs```

**Comment by lbalmaceda** [comment]
@minhlongdo because we still support JDK 7 you need to change this line https://github.com/auth0/auth0-java/pull/120/files#diff-a5e9c9a379e91d2a7b2790bae000ed6bR35 and cast the second parameter to `Object`, like we do in many other tests.

```
assertThat(body, hasEntry("user_id", (Object) "google-oauth2|1234"));
```

**Comment by minhlongdo** [comment]
@lbalmaceda ok thank you for pointing it out, I am unable to test it with JDK 7 :(

**Comment by minhlongdo** [comment]
@lbalmaceda Do you want me to do some squashing?

**Comment by lbalmaceda** [comment]
I'll review this PR again tomorrow. You can squash them if you want or I can do that myself when I merge it. ⚡️

File: src/main/java/com/auth0/client/mgmt/JobsEntity.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        Map<String, String> requestBody = new HashMap<>();
+        requestBody.put("user_id", userId);
+        if (clientId != null && !clientId.isEmpty()) {
+            requestBody.put("client_id", clientId);
+        }
+
+        CustomRequest<Job> request = new CustomRequest<>(client, url, "POST", new TypeReference<Job>() {
+        });
+        request.addHeader("Authorization", "Bearer " + apiToken);
+        request.setBody((Object) requestBody);
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/JobsEntity.java]: this one doesn't need to be casted to `Object`.

File: src/main/java/com/auth0/client/mgmt/ManagementAPI.java [hunk 5]
```diff
@@ -231,4 +231,13 @@ public TicketsEntity tickets() {
     public ResourceServerEntity resourceServers() {
         return new ResourceServerEntity(client, baseUrl, apiToken);
     }
+
+    /**
+     * Getter for Jobs entity.
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/ManagementAPI.java]: Getter for _the_ Jobs entity

File: src/test/java/com/auth0/client/mgmt/JobsEntityTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        Map<String, Object> body = bodyFromRequest(recordedRequest);
+        assertThat(body.size(), is(1));
+        assertThat(body, hasEntry("user_id", (Object) "google-oauth2|1234"));
+
+        assertThat(response, is(notNullValue()));
+    }
+
+    @Test
+    public void shouldSendUserAVerificationEmailWithClientId() throws Exception {
+        Request<Job> request = api.jobs().sendVerificationEmail("google-oauth2|1234", "google-oauth2|client-id");
```
Comment (lbalmaceda) [inline src/test/java/com/auth0/client/mgmt/JobsEntityTest.java]: I know this is just a test, but a client id value looks different. Try something similar to `AaiyAPdpYdesoKnqjj8HJqRn4T5titww` as used in another test please.

File: src/test/java/com/auth0/client/mgmt/JobsEntityTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        Map<String, Object> body = bodyFromRequest(recordedRequest);
+        assertThat(body.size(), is(2));
+        assertThat(body, hasEntry("user_id", (Object) "google-oauth2|1234"));
+        assertThat(body, hasEntry("client_id", (Object) "google-oauth2|client-id"));
+
+        assertThat(response, is(notNullValue()));
+    }
+
+    @Test
+    public void shouldThrowOnNullEmailRecipient() throws Exception {
```
Comment (lbalmaceda) [inline src/test/java/com/auth0/client/mgmt/JobsEntityTest.java]: Depending on what you choose to fix the other comment, this test might need to be renamed.

**Pull request merged**